### PR TITLE
Fix #117, unescape URI-encoded paths for icon files

### LINKED ISSUE #117: Can't handle image paths containing spaces [$10]
Sending a notification (with app_icon, image_path and image-path all set to the same uri) with a file URI for an image, derived from a path containing spaces fails to load.

e.g. "file:///somepath/my%20image.png" results in mate-notification-daemon trying to load "/somepath/my%20image.png" while it should be "/somepath/my image.png". At least this is what I see using strace.

I've tested this on Ubuntu 16.10 which has version "1.16.0-1"

<bountysource-plugin>

---
There is a **[$10 open bounty](https://www.bountysource.com/issues/40791743-can-t-handle-image-paths-containing-spaces?utm_campaign=plugin&utm_content=tracker%2F1172660&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1172660&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by AlfioEmanueleFresta**
Fixes issue #117, which caused file:// URIs containing correctly-escaped special characters, such as spaces (`%20`), not to be resolved.

### CODE REVIEW

**Comment by raveit65** [comment]
How can this be tested?

**Comment by raveit65** [comment]
Btw.
notify-send /home/rave/Bilder/%20image.jpg
displays me the link.
or
notify-send -i /home/rave/Bilder/%20image.jpg test
displays a test notification with an icon inside.
@AlfioEmanueleFresta 
What is wrong without your PR?

**Comment by raveit65** [comment]
hmm,
notify-send -i /home/rave/Bilder/my image.jpg test
doesn't display the image
Here *image* is used for header and test is the *body*.
But with your PR i see no different.
So, please provide some test instructions.
Of course this one works well.
notify-send -i /home/rave/Bilder/my\ image.jpg test

**Comment by AlfioEmanueleFresta** [comment]
@raveit65, in normal paths, `%20` should not be replaced with a space character. This is because the percentage sign is allowed in file names, and you may legitimately have a filename with `%20` in it. In your case, you should be able to normally load an image named `%20image.jpg`, but if you rename your image to have a leading space character in the file name, e.g,
```
/home/rave/Bilder/ image.jpg
```
you should *not* be able to access it by referring it as,
```
/home/rave/Bilder/%20image.jpg
```

However, file:// URIs may have URL-encoded characters (as per [RFC 1738](https://tools.ietf.org/html/rfc1738#section-2.2)), and these need to be decoded. In your case, you should be able to access your image with a whitespace in the filename,
```
/home/rave/Bilder/ image.jpg
```
by using the following file URI,
```
file:///home/rave/Bilder/%20image.jpg
```

In conclusion, to test this PR, you may create an image path with spaces in it, e.g,
```
/home/rave/path with spaces/test image.jpg
```
then,
```bash
notify-send -i "file:///home/rave/path%20with%20spaces/test%20image.jpg" "Summary"
```
should correctly load the image.

I hope this clarifies this issue.

**Comment by raveit65** [comment]
Yeap, that works
without your PR
```
notify-send -i "file:///home/rave/Bilder/path%20with%20spaces/my%20image.jpg" "Summary"
```
doesn't display the image.
With your PR it works.

**Comment by raveit65** [comment]
merged and reworded commit message a bit.
https://github.com/mate-desktop/mate-notification-daemon/commit/0a84eac2c5115d6a21e98e4797fbf6407cf5712b
The hint 
```
Fix #117
```
belongs to commit body and not in header ;-)

**Comment by raveit65** [comment]
...thank you.
